src/main/java/org/checkerframework/languageserver/CFTextDocumentService.java

```diff
@@ -163,4 +171,34 @@ public void publish(Map<String, List<javax.tools.Diagnostic<?>>> result) {
                                     .collect(Collectors.toList())));
         }
     }
+
+    /**
+     * The hover request is sent from the client to the server to request hover information at a
+     * given text document position.
+     *
+     * <p>Registration Options: TextDocumentRegistrationOptions
+     */
+    @Override
+    public CompletableFuture<Hover> hover(HoverParams params) {
+        generateTestData();
+        // line 1 on vscode is 0
+        int line = params.getPosition().getLine();
+        logger.info(params.toString());
+
+        if (typeRefinementMapping.get(line) != null) {
+            Hover result =
+                    new Hover(
+                            new MarkupContent(
+                                    MarkupKind.PLAINTEXT, typeRefinementMapping.get(line)));
+            return CompletableFuture.completedFuture(result);
+        }
+
+        return CompletableFuture.completedFuture(null);
```

Comment:
Could you please try enabling this feature and check how the data is shown in client?

Reply:
When this is enabled the lines that are not contained within the rangemap do not display anything when hovered over